learnify/mobile/lib/features/auth/verification/view-model/verification_view_model.dart

```diff
@@ -65,6 +79,41 @@ class VerificationViewModel extends BaseViewModel {
   }
 
   Future<String?> _verificationRequest() async {
+    final String verificationCode = _verificationCodeFirstDigitController.text +
+        _verificationCodeSecondDigitController.text +
+        _verificationCodeThirdDigitController.text +
+        _verificationCodeFourthDigitController.text;
+
+    final VerifyEmailRequest requestModel =
+        VerifyEmailRequest(email: email, code: verificationCode);
+
+    final IResponseModel<VerifyEmailResponse> resp =
+        await _authService.verifyEmail(requestModel);
+
+    if (resp.hasError) return resp.error?.errorMessage;
+    await navigationManager.navigateToPage(
+        path: NavigationConstants.home,
+        data: <String, dynamic>{
+          "token": resp.data?.token,
+          "user": resp.data?.user
```

Comment:
You cannot pass a model like this!!! This will crash the app. You should always convert it to a map like "resp.data?.user.toJson", I fixed it instead of you, please don't forget this.